pkgs/os-specific/linux/kernel/patches.nix

```diff
@@ -57,4 +57,62 @@ rec {
+      name = args.name;
+      patch = fetchpatch {
+        name   = "linux-ck-4-14-${args.fn}";
+        url    = "http://ck.kolivas.org/patches/4.0/4.14/4.14-ck1/patches/${args.fn}";
+        sha256 = args.h;
+      };
+    };
+
+    topatches = list: map topatch list;
+  in
+  topatches [
+    { fn = "0001-MuQSS-version-0.162-CPU-scheduler.patch";                    h = "0cy8h00asjmispdcaj0gpd5jc1hqw32fhdm44s5b88w7jm6gwpy9"; }
+    { fn = "0002-Make-preemptible-kernel-default.patch";                      h = "147wj40f895xfjvb8ihmzhyf1bdbmll8jqrkpgg48r71hxv48rka"; }
+    { fn = "0003-Expose-vmsplit-for-our-poor-32-bit-users.patch";             h = "1dwpypwq77wy81zinpr8k5jj27js0pmd4xss9mh7cvzzbhdg55xq"; }
+    { fn = "0004-Create-highres-timeout-variants-of-schedule_timeout-.patch"; h = "1lr45rg991l6jp3a21hqvnhjbwzndi6w0jgdm0jx5g5gyw8xiad3"; }
+    { fn = "0005-Special-case-calls-of-schedule_timeout-1-to-use-the-.patch"; h = "1jf7r4y81a64r2vd9dn8ga2b6561rjyk684hn2gz207b8k2h91f6"; }
+    { fn = "0006-Convert-msleep-to-use-hrtimers-when-active.patch";           h = "0hpy7c1f5zn9drijdplc901x1xn404dm7zhz1jqabwj82qpxq5aq"; }
+    { fn = "0007-Replace-all-schedule-timeout-1-with-schedule_min_hrt.patch"; h = "126g146607zpg4gri116f834k4079yqgychxqlp4k3fqi4q2bl0i"; }
+    { fn = "0008-Replace-all-calls-to-schedule_timeout_interruptible-.patch"; h = "1yszs4iw5scbg7xvwmi22nxgbda0wk8nnbjjixr9aphmggxi2fqw"; }
+    { fn = "0009-Replace-all-calls-to-schedule_timeout_uninterruptibl.patch"; h = "13058jvnf1vld68a9gad21randsc5fv8xc09z2vkb9480w6x8pnr"; }
+    { fn = "0010-Don-t-use-hrtimer-overlay-when-pm_freezing-since-som.patch"; h = "1lr9wy2xjsq7w0yvmk9zf5qi3a10zdll2z4b2k1ma4hfjqri9lms"; }
+    { fn = "0011-Make-hrtimer-granularity-and-minimum-hrtimeout-confi.patch"; h = "13jgl6b0gcxpvklp6hd8mds2mhnzspjla61y0x3j2yqripfnd6q9"; }
+    { fn = "0012-Reinstate-default-Hz-of-100-in-combination-with-MuQS.patch"; h = "0wxlx702y2y9f2p7gb9f0alrirclfn7w1s0j793giaixnn934dgh"; }
+    { fn = "0013-Make-threaded-IRQs-optionally-the-default-which-can-.patch"; h = "11a6bnzx8am3zzbma10qim7z44zr8kzfwzx4p3kb83b3kc3lxgs9"; }
+    { fn = "0014-Swap-sucks.patch";                                           h = "16saq7s6bq88a439fpdavdd86ysrv5g63b80r1xc5w0qx9dhk3yf"; }
+    { fn = "0015-Enable-BFQ-io-scheduler-by-default.patch";                   h = "0blz2in6vp6vcdpmpxlw6bh5c5gjca552xz0izfmxnwfrxwvgiq9"; }
+    { fn = "0016-Add-ck1-version.patch";                                      h = "1k1dd3bxm3vf6afzbgw64ddmsng0w4w9l0yg6kb3fz82l22cwpc8"; }
+  ];
+
+  ck_4_19 = let
+    unpackedPatch = stdenv.mkDerivation rec {
+      name = "4.19-ck1";
+
+      src = fetchurl {
+        url    = http://ck.kolivas.org/patches/4.0/4.19/4.19-ck1/patch-4.19-ck1.xz;
+        sha256 = "77863d16a08e1b3c726b6c965f1bb7c672bd7317776810121062b73f9ea26780";
+      };
+
+      unpackPhase = "true";
+      installPhase = ''
+        unxz -c $src > $out
+      '';
+    };
+  in rec {
+    patch = unpackedPatch;
```

Comment:
Unfortunately (@worldofpeace) does the patchset not apply on the kernel sources.

I'm not sure why, though. Maybe the kernel is too new (as in patchlevel-version)?

Reply:
```
[...]
Hunk #1 succeeded at 155 (offset 24 lines).
patching file sound/usb/line6/pcm.c
patching file tools/objtool/Makefile
Hunk #1 FAILED at 31.
1 out of 1 hunk FAILED -- saving rejects to file tools/objtool/Makefile.rej
builder for '/nix/store/qb8k5whqxapcb3yknyqssfrfqpljif9p-linux-config-4.19.34.drv' failed with exit code 1                                                                                                       
cannot build derivation '/nix/store/lsdl2wy98a73d1s7rvl4qknnjp2s2ha6-linux-4.19.34.drv': 1 dependencies couldn't be built 
```

Reply:
Hmm, I don't think the author keep these patches in sync for very long.
So we might only be able to do patches for the latest, or fetch from places that have corrected the patch so it can apply.